src/daemon.cpp

```diff
@@ -27,6 +27,20 @@ Daemon::~Daemon() { MVPN_COUNT_DTOR(Daemon); }
 bool Daemon::activate(const Config& config) {
```

Comment:
The Daemon is shared between macos, windows and linux. Each daemon behaves at the same way and it has the following functionalities:

1. activate/deactivate the VPN
2. retrieve the current status
3. retrieve the logs

With this patch we are adding a new feature: server-switching. Not all the daemon impls are able to do this operation in 1 single step (yet). If they are not able to (`supportServerSwitching()`), calling `activate` when connected, ends up in a `deactivate` + `activate`.

We have different daemon implementation because each platform has a different way to run it.
- windows: Windows Service
- macos: unix socket
- linux: dbus

For each daemon impl, we have a different controller. I managed to unify windows and mac in a single impl: `LocalSocketController`. Linux has its own `LinuxController`. Maybe we can rewrite `LinuxController` on top of `LocalSocketController`, but this is not related with this patch.

To answer your question, `activate` can be called at any time. Usually they are called because of a user interaction.